cisc/cisc.go

```diff
@@ -387,6 +410,79 @@ func scRoster(c *cli.Context) error {
 	return nil
 }
 
+func scRosterAdd(c *cli.Context) error {
+	cfg := loadConfigOrFail(c)
+	id, err := cfg.findSC(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+	if id == nil {
+		scList(c)
+		return errors.New("Please chose one of the existing skipchain-ids")
+	}
+
+	si := getServerIdentity(c.Args().Get(0))
+	prop := id.GetProposed()
+	prop.Roster = onet.NewRoster(append(id.Roster().List, si))
+	cfg.proposeSendVoteUpdate(id, prop)
+	log.Info("Proposed new roster for skipchain")
+	if id.Proposed == nil {
+		log.Info("New roster has been accepted")
+	}
+	return nil
+}
+
+func scRosterRemove(c *cli.Context) error {
+	cfg := loadConfigOrFail(c)
+	id, err := cfg.findSC(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+	if id == nil {
+		scList(c)
+		return errors.New("Please chose one of the existing skipchain-ids")
+	}
+
+	si := getServerIdentity(c.Args().Get(0))
+	prop := id.GetProposed()
+	roster := onet.NewRoster(id.Roster().List)
+	index := -1
+	for i, s := range roster.List {
+		if s.Equal(si) {
+			index = i
+		}
+	}
+	if index == -1 {
+		return errors.New("Couldn't find this node in the roster")
+	}
+	roster.List = append(roster.List[0:index], roster.List[index+1:]...)
+	prop.Roster = roster
+	cfg.proposeSendVoteUpdate(id, prop)
+	log.Info("Proposed new roster for skipchain")
+	if id.Proposed == nil {
+		log.Info("New roster has been accepted")
+	}
+	return nil
+}
+
+func getServerIdentity(fileIP string) *network.ServerIdentity {
+	if _, err := os.Stat(fileIP); err == nil {
+		// Suppose it's a roster file
+		g, err := getGroupString(fileIP)
```

Comment:
There is a possibility that the file exists but has a name in the format of "ip:port". There may be some confusion because the user might expect that the "ip:port" to be used directly rather than what is in the file. I'd prefer to use explicit flags for these kind of options, e.g. `--toml` and `--addr`.

Reply:
Muggle alarm! Magicians like magic...